cva6/tests/custom/sv32/vm_mxr_unset_01.S

```diff
@@ -0,0 +1,157 @@
+#=======================================================================
+#  Load Access of Level1 PTE in User and Supervisor mode when 
+#  mstatus.MXR is unset 
+#-----------------------------------------------------------------------
+# Test Description:
+#
+# If PTE has only execute permission (pte.x = 1) and s/mstatus.MXR=0, 
+# then read access on that PTE should raise load page fault exception.
+#
+# - Set pte.r=0 & pte.w=0 & pte.x=1 & s/mstatus.MXR=0 and test the 
+#   read acces.
+#
+#=======================================================================
+
+#include "macros.h"
+#define _MMODE_ "M"
+#define _SUMODE_ "SU"
+
+#ifdef smode
+    #define SET_PTE_U 0
+#else
+    #define SET_PTE_U PTE_U
+#endif
+
+.text
+.global _start
+.option norvc
+
+_start:
+    ALL_MEM_PMP                                                         # PMP permission to all the mem
+    la t1,trap_handler                                                  # loads the address of trap handler 
+    csrw mtvec,t1                                                       # sets the mtvec to trap handler 
+    
+PTE_LEVEL1_SUPERVISOR:
+
+# ----------------LEVEL 1 PTE Setup for load and store test---------------------
+
+    la a1,vm_en                                                         # loads the address of label vm_en
+    mv a0, a1                                                           # VA = PA - Identity Map
+    ori a2, x0, ( PTE_D | PTE_A | SET_PTE_U | PTE_X | PTE_V )           # sets the permission bits
+    PTE_SETUP_RV32(a1, a2, t1, a0, pgtb_l1, LEVEL1)                     # setup the PTE for level1
+ 
+    la a1,rvtest_data                                                   # loads the address of label rvtest_data
+    mv a0, a1                                                           # VA = PA - Identity Map
+    ori a2, x0, ( PTE_D | PTE_A | SET_PTE_U | PTE_X | PTE_W | PTE_V )   # sets the permission bits
```

Comment:
We only need to set pte.x for this test, right?
```suggestion
    ori a2, x0, ( PTE_D | PTE_A | SET_PTE_U | PTE_X | PTE_V )   # sets the permission bits
```

Reply:
Done